CMakeLists.txt

```diff
@@ -133,6 +133,44 @@ endif()
 find_package(Threads REQUIRED)
 find_package(ZLIB REQUIRED)
 
+# try to find an installed boost
+set(boost_VERSION "1.71")
+find_package(Boost ${boost_VERSION} QUIET)
+if (NOT Boost_Found)
+  # bail if there's no conan installed
+  message(STATUS "No compatible boost version detected, using conan...")
+  find_program(conan_FOUND conan)
+  if (NOT conan_FOUND)
+    message(FATAL_ERROR "conan needs to be installed for boost, see docs.conan.io/en/latest/installation.html.")
+  endif()
+
+  if(NOT EXISTS "${CMAKE_BINARY_DIR}/conan.cmake")
+    message(STATUS "Valhalla: Downloading conan.cmake from https://github.com/conan-io/cmake-conan")
+    file(DOWNLOAD "https://raw.githubusercontent.com/conan-io/cmake-conan/0.17.0/conan.cmake"
+                  "${CMAKE_BINARY_DIR}/conan.cmake"
+                  EXPECTED_HASH SHA256=3bef79da16c2e031dc429e1dac87a08b9226418b300ce004cc125a82687baeef
+                  TLS_VERIFY ON)
+  endif()
+
+  list(APPEND CMAKE_MODULE_PATH ${CMAKE_BINARY_DIR})
+  list(APPEND CMAKE_PREFIX_PATH ${CMAKE_BINARY_DIR})
+
+  include(${CMAKE_BINARY_DIR}/conan.cmake)
+
+  conan_cmake_configure(REQUIRES boost/${boost_VERSION}.0
+    GENERATORS cmake_find_package
+    OPTIONS boost:header_only=True)
+  conan_cmake_autodetect(settings)
+  conan_cmake_install(PATH_OR_REFERENCE .
+                      INSTALL_DIR ${VALHALLA_SOURCE_DIR}/test_conan
+                      REMOTE conancenter
+                      SETTINGS ${settings}
+                      OUTPUT_QUIET)
+
+  find_package(Boost ${boost_VERSION})
```

Comment:
the `boost:header_only=True` is still pulling the headers for all libraries, incl the ones which need linking. in total it's ~160 MB (down from multiple GB in git submodules).

for future ref @mloskot : this is apparently what conan [recommends](https://github.com/conan-io/cmake-conan#conan_cmake_run-high-level-wrapper) these days, also doesn't need a `conanfile.txt`, but a few more lines.

Reply:
@nilsnolde The link points to the `conan_cmake_run() high level wrapper` section which says _"This function is **not** the recommended way of using cmake-conan any"_. Do you mean that section ?

Reply:
yeah right they recommend using the single functions like `conan_cmake_configure()` etc further up in that document. was more of a FYI, maybe avoids some confusion if your boost.gil conan is not functioning properly at some point.

Reply:
Yes, thanks for the update. I'll have to catch the GIL scripts up.